lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -1762,9 +1762,12 @@ void ReabstractionInfo::finishPartialSpecializationPreparation(
 /// This constructor is used when processing @_specialize.
 ReabstractionInfo::ReabstractionInfo(ModuleDecl *targetModule,
                                      bool isWholeModule, SILFunction *Callee,
-                                     GenericSignature SpecializedSig)
-    : TargetModule(targetModule), isWholeModule(isWholeModule) {
-  Serialized = Callee->isSerialized();
+                                     GenericSignature SpecializedSig,
+                                     bool isPrespecialization)
+    : TargetModule(targetModule), isWholeModule(isWholeModule),
+      isPrespecialization(isPrespecialization) {
+  Serialized =
+      this->isPrespecialization ? IsNotSerialized : Callee->isSerialized();
```

Comment:
Why can't we serialize the pre-specialization if the original function is serialized?

Reply:
This is intentional. Pre-specialization as implemented in this patch get their own symbol/mangling to make them easily identifiable for tooling. The intent of exported pre-specializations is to reduce code size. When using the pre-specializated symbol I don't want the code to be available in the client module for further optimization.